File: src/main/java/com/gdosoftware/biblioteca/domain/interfaces/ISocioCasoUso.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.gdosoftware.biblioteca.domain.interfaces;

import com.gdosoftware.biblioteca.domain.modelo.Socio;
import java.util.List;

/**
 *
 * @author Dani
 */
public interface ISocioCasoUso {
    public Socio createLibro(Socio libro);
    public Socio retriveLibro(Long id);
    public Socio updateLibro(Long id, Socio libro);
    public void deleteLibro(Long id);
    public List<Socio> findAllLibro();
}

File: domain/interfaces/iSocioCasoUso.go
package interfaces

import "github.com/gdosoftware/biblioteca/domain/modelo"

type ISocioCasoUso interface {
	CreateSocio(socio *modelo.Socio) (*modelo.Socio, error)
	RetrieveSocio(id string) (modelo.Socio, error)
	UpdateSocio(id string, socio *modelo.Socio) (*modelo.Socio, error)
	DeleteSocio(id string) error
	FindAllSocio() ([]modelo.Socio, error)
}
